src/app/utils.ts

```diff
@@ -347,6 +351,55 @@ export function detectBrowserLanguage(defaultLanguage: string = "en"): string {
   return defaultLanguage;
 }
 
+// Gets amount precision for each token traded on dexteronradix.
+// Note: precision for price is different.
+export function getPrecision(input: string): number {
+  return (
+    {
+      XRD: 2,
+      DEXTR: 2,
+      CASSIE: 0,
+      CAVIAR: 2,
+      DFP2: 2,
+      DGC: 0,
+      FLOOP: 5,
+      HUG: 0,
+      MNI: 0,
+      OCI: 2,
+      PLANET: 0,
+      RDK: 0,
+      WEFT: 2,
+      XRAW: 0,
+      XUSDC: 2,
+    }[input.toUpperCase()] || 2
+  );
+}
+
+export function formatNumericString(
+  value: string,
+  separator: string,
+  scale: number
+): string {
+  const regex = separator === "." ? /[^\d.-]/g : /[^\d,-]/g;
+  let formattedValue = value.replace(regex, "");
+  // Ensure only the first occurrence of the separator is allowed
+  const parts = formattedValue.split(separator);
+  if (parts.length > 2) {
+    // Rejoin with a single separator, discarding additional separators
+    formattedValue = parts[0] + separator + parts.slice(1).join("");
+  }
+  // Allow a trailing separator for user input
+  if (formattedValue.endsWith(separator)) {
+    return formattedValue;
+  }
+  // Split and limit fraction scale as before
+  let [whole, fraction] = formattedValue.split(separator);
+  if (fraction && fraction.length > scale) {
+    fraction = fraction.substring(0, scale);
+  }
+  return fraction ? `${whole}${separator}${fraction}` : whole;
+}
```

Comment:
it would be great to have unit tests for this function

Reply:
Done!